susemanager-utils/susemanager-sls/src/tests/test_beacon_pkgset.py

```diff
@@ -22,12 +22,12 @@ def test_validate():
     '''
     Test validate() function
     '''
-    res, msg = pkgset.validate({'cookie': '/bogus/path'})
-    assert res is True
-    assert msg == 'Configuration validated'
-
-    for cfg in [{}, {'bogus': 'data'}]:
-        res, msg = pkgset.validate(cfg)
+    with patch.object(pkgset.os.path, "exists", MagicMock(return_value=True)):
+        res, msg = pkgset.validate({})
```

Comment:
I didn't see the original PR until now, but since `config` is now unused, I think it shouldn't be removed from `pkgset.validate`'s  API.

Reply:
Or is this part of Salt's api? Then I'd rename it to `_config`, but no need to change anything here.

Reply:
There is no need to check the config now, the only `interval` value is make sense now, but it's managed outside the exact beacon. The test unit previously just contains the checks for empty config or confing which doesn't contain `cookie`. In current state the only important point if the plugin itself is available on the system or not.

Reply:
I understand that. The "issue" I have is that it is unclear why the function `validate` accepts a parameter `config` which it then ignores. If Salt itself passes a `dict` to `validate()` as part of its beacon API, we can't remove `config` from the function definition. In that case, I would mark it as "a parameter that is expected to be ignored", either by prefixing it with an `_` or by "deleting" it as the first thing in the function body:

```python
def validate(config):
    del config # passed by Salt, not needed in validate
    ...
```
The upside of doing this is that it is clear that `config` was not forgotten and linters don't complain.


If the `config` parameter is not required by Salt, I would remove it.

Reply:
I see what you mean. Will fix it in the next PR. Most probably need to change the messages as current messages in validate is not reflecting real meaning.